devices/src/gic.rs

```diff
@@ -23,6 +26,22 @@ type Result<T> = result::Result<T, Error>;
 pub const IRQ_LEGACY_BASE: usize = layout::IRQ_BASE as usize;
 pub const IRQ_LEGACY_COUNT: usize = 32;
 
+/// Default config implied by layout::*
+pub fn create_vgic_config(vcpu_count: u64) -> VgicConfig {
```

Comment:
Woulndn't `VgicConfig::new()` be more appropriate

Reply:
I can't implement VgicConfig::new() here, since it's in a different crate from where VgicConfig is defined

Reply:
It would make make most sense for them to be together, no?

Reply:
VgicConfig has to be defined in the hypervisor crate (which makes sense - along with the Vgic trait).
But the hypervisor crate can't access arch::layout or devices::gic so I put create_vgic_config here...

I tried making the hypervisor crate depend on the arch crate but it creates a cyclic dependency.
Making devices crate depend on hypervisor crate also creates a cycle...

I'm not sure how to put VgicConfig and its initialization together and have them accessible from inside and outside the hypervisor crate!